src/control/mod.rs

```diff
@@ -341,7 +341,10 @@ pub trait Device: super::Device {
         handle: framebuffer::Handle,
         clips: &[ClipRect],
     ) -> Result<(), SystemError> {
-        ffi::mode::dirty_fb(self.as_fd().as_raw_fd(), handle.into(), clips)?;
+        ffi::mode::dirty_fb(self.as_fd().as_raw_fd(), handle.into(), unsafe {
+            // SAFETY: ClipRect is repr(transparent) for drm_clip_rect
+            core::slice::from_raw_parts(clips.as_ptr() as *const ffi::drm_clip_rect, clips.len())
```

Comment:
Since we do now have bytemuck in our dependencies to handle these unsafe operations, would you mind replacing this with a `cast_slice`-call? https://docs.rs/bytemuck/latest/bytemuck/fn.cast_slice.html

To elaborate, I don't think your code is unsafe, but we already shifted that responsibility to bytemuck in other cases, so if we can avoid added unsafe code ourselves here by further relying on that crate, I would like to do that.

Reply:
We can't use `bytemuck` here as `drm_clip_rect` is not marked as `Pod`.